Sources/App/Views/PackageController/PackageShowView.swift

```diff
@@ -191,3 +140,107 @@ class PackageShowView: PublicPage {
     }
 
 }
+
+
+extension PackageShowView {
+    struct Model {
+        let title: String
+        let url: URL
+        let license: License
+        let summary: String
+        let authors: [Link]
+        let history: History
+        let activity: Activity
+        let products: (libraries: Int, executables: Int)
+
+        struct Link {
+            let name: String
+            let url: URL
+        }
+
+        struct History {
+            let since: String  // TODO: use Date and derive on the fly
+            let commits: Link
+            let releases: Link
+        }
+
+        struct Activity {
+            let openIssues: Link
+            let pullRequests: Link
+            let lastPullRequestClosedMerged: String
+        }
+    }
+}
+
+
+extension PackageShowView.Model {
+    func authorsClause() -> [Node<HTML.BodyContext>] {
+        switch authors.count {
+            case 0:
```

Comment:
I was planning to pull this functionality out into a helper class. There's a lot of things on this page that need making into a comma-separated sentence. The logic is good though 👍

Reply:
Yes, please don't hesitate to move things around. I've just picked a couple of file and type names as they seemed to make sense right now but I image we'll need to reorganise that a bit as it grows.